src/components/RequestBox/RequestBox.js

```diff
@@ -40,6 +40,24 @@ export default class RequestBox extends Component {
     this.renderError = this.renderError.bind(this);
   }
 
+  replaceRequestAndSubmit(request) {
+    console.log("replaceRequestAndSubmit");
+
+    // replace the request in the state with the new one
+    if (request.resourceType.toUpperCase() === "MEDICATIONREQUEST") {
+      this.state.medicationRequest = request;
```

Comment:
we shouldn't be setting the state of a react component directly, should use `this.setState({ medicationRequest: request });`. Same for the others below.

Reply:
updated